### PR TITLE
MySQL: Add parseTime=true to SQL connections

### PR DESCRIPTION
**Pull request by ryantxu**
**What is this feature?**

This adds a feature flag to set the `parseTime=true` flag on the golang mysql driver.  

**Why do we need this feature?**

This is required for [zanzana](https://github.com/grafana/grafana/blob/main/pkg/services/authz/zanzana/store/store.go#L53) and sqlx date parsing.

**Who is this feature for?**

sqlx developers and zanzana

-----

Ideally this feature flag can be very quickly removed after verifying that everything works as expected in cloud.

### CODE REVIEW

File: pkg/services/sqlstore/database_config.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ func (dbCfg *DatabaseConfig) buildConnectionString(cfg *setting.Cfg, features fe
 			cnnstr += fmt.Sprintf("&transaction_isolation=%s", val)
 		}
 
+		if features != nil && features.IsEnabledGlobally(featuremgmt.FlagMysqlParseTime) {
```
Comment (ryantxu) [inline pkg/services/sqlstore/database_config.go:169]: https://github.com/go-sql-driver/mysql?tab=readme-ov-file#parsetime

**Review by chaudyg — APPROVED** [review]

**Comment by ryantxu** [comment]
Yes we can test this a bit more to make sure it works as expected, but from what I have tried it defautls to UTC see:
https://github.com/go-sql-driver/mysql/blob/master/README.md#timetime-support

and importantly the loc defaults to UTC https://github.com/go-sql-driver/mysql/blob/master/README.md#loc

File: pkg/services/sqlstore/sqlstore.go [hunk 2]
```diff
@@ -296,15 +296,24 @@ func (ss *SQLStore) initEngine(engine *xorm.Engine) error {
 		}
 	}
 	if engine == nil {
+		connection := ss.dbCfg.ConnectionString
+
+		// Ensure that parseTime is enabled for MySQL
+		if ss.dbCfg.Type == migrator.MySQL && !strings.Contains(connection, "parseTime=") {
+			if ss.features.IsEnabledGlobally(featuremgmt.FlagMysqlParseTime) {
```
[the remark this answers is not in the document]
Comment (alexanderzobnin) [reply pkg/services/sqlstore/sqlstore.go:303]: Yeah, I've got the same

[the remark this answers is not in the document]
Comment (ryantxu) [reply pkg/services/sqlstore/sqlstore.go:303]: https://github.com/grafana/grafana/pull/92484

**Pull request merged**